### ISSUE TITLE
Design: Chat name+color change modal

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#2205]
### Share your bug report, feature request, or comment.

This modal needs a design.

![image](https://user-images.githubusercontent.com/414923/195481176-98283c21-23df-4d40-a34b-7db4cd73a6a2.png)

Component: https://owncast.online/components/?path=%2Fstory%2Fowncast-modals-name-change--basic

**Comment by xarantolus** [comment]
Hey, I'm interested in working on this :)
Apart from the name & color, which other info should go into the dialog?

**Comment by gabek** [comment]
Hi there, thanks so much for wanting to help!

The only two inputs like you  mentioned are name and color, but I think it might be useful to mention the chat authentication that's offered as well, since those two things are both chat user related and somebody doing one of those things might be interested in the other. At one point I thought about merging all of that into one dialog (See https://owncast.online/components/?path=%2Fstory%2Fowncast-modals-auth--basic for the auth modal) but since display characteristics and the act of going through the auth flow are different I left them separate. 

Feel free just to put some placeholder text somewhere and we can tweak it over time. 

Thanks again, and do ask any other questions that might come to mind!

<blockquote><div><strong><a href="https://owncast.online/components/?path=%2Fstory%2Fowncast-modals-auth--basic">Webpack App</a></strong></div></blockquote>

**Comment by xarantolus** [comment]
Thank you for the info! :) 
I tried designing this on my branch [here](https://github.com/xarantolus/owncast/tree/feature/name-change-design) and it now looks like this:


https://user-images.githubusercontent.com/32465636/201861848-22408a12-ffee-419c-b174-f299350e42ee.mp4


Two things:
* I think the round part within the text input looks a bit weird, not sure if this design is intentional.
* The color selector doesn't really work on smaller screens, there it kind of stays within the modal, which makes it impossible to click some colors. I tried setting a z-index and similar but didn't get it to "overflow" the modal
